### ISSUE TITLE
Remote config redirects do not resolve relative Location values

### ISSUE THREAD

**Issue by OskarEichler** [rubocop/rubocop#15626]
## Expected behavior

Remote configuration should follow a valid relative HTTP redirect by resolving the `Location` header against the current config URI.

## Actual behavior

`RuboCop::RemoteConfig#request` parses the header as a standalone URI. A response such as `Location: ../shared/base.yml` therefore produces a URI without a host, and the next request raises `NoMethodError` while trying to open the connection.

## Reproduction

A local server returns:

```
GET /configs/project.yml
302 Location: ../shared/base.yml
```

and serves valid RuboCop YAML at `/shared/base.yml`.

RuboCop 1.90.0 and current master fail after the first request instead of loading the redirected configuration. The model reproduces on Ruby 4.0.6 and Ruby 3.2.11.

## Suggested direction

Resolve the redirect relative to the current request URI:

```ruby
request(URI.join(uri, response['location']), limit - 1, &block)
```

An isolated source-only candidate passes the existing 27 remote-config examples on both Rubies, all 943 runtime files compile on both, changed-file lint passes, and the complete current suite passes 33,206 examples with three existing pending examples on Ruby 4.0.6.

I searched the current open issue/PR set and found no overlap. A full contribution would still need the project-required regression spec and changelog entry.

**Comment by RedZapdos123** [comment]
Opened a PR to resolve the issue.
